desktop-src/direct3dhlsl_current/hlsl-method-gatherCmp.md

```diff
@@ -0,0 +1,283 @@
+# GatherCmp Method
+
+Gets the four samples that would be used for bilinear interpolation when sampling a texture.
+
+Method Signatures vary per supported TextureObject type:
+
+### Signatures
+
+#### Texture2D
+```syntax
+// same as GatherRed
+ElementType Texture2D<ElementType>::GatherCmp(
+      in  SamplerState  samp,
+      in  float2        coord,
+      in  float         compareValue,
+      in  int2          offset,
+   [, out uint          status ] );
+
+ElementType Texture2D<ElementType>::GatherCmpRed(
+      in  SamplerState  samp,
+      in  float2        coord,
+      in  float         compareValue,
+      in  int2          offset,
+   [, in  int2          offset2]
+   [, in  int2          offset3]
+   [, in  int2          offset4]   
```

Comment:
Offset2/3/4 aren't independently optional.  You can either use one offset, or 4 offsets.  So perhaps the syntax here should look more like this:
```
   [, in  int2          offset2
    , in  int2          offset3
    , in  int2          offset4]
```

Reply:
Oh and same for all similar places here and for normal Gathers

Reply:
I will also include in the introduction that the signatures are valid only with the first offset value or all 4.  There are no other signatures.